doc/src/devdocs/llvm-passes.md

```diff
@@ -0,0 +1,159 @@
+
+This pass lowers exception handling intrinsics into calls to runtime functions that are actually called when handling exceptions.
+
+### RemoveNI
+
+* Filename: `llvm-remove-ni.cpp`
+* Class Name: `RemoveNIPass`
+* Opt Name: `module(RemoveNI)`
+
+This pass removes the non-integral address spaces from the module's datalayout string. This enables the backend to lower Julia's custom address spaces directly to machine code, without a costly rewrite of every pointer operation to address space 0.
+
+### SIMDLoop
+
+* Filename: `llvm-simdloop.cpp`
+* Class Name: `LowerSIMDLoopPass`
+* Opt Name: `module(LowerSIMDLoop)`
+
+This pass acts as the main driver of the `@simd` annotation. Codegen inserts a call to a marker intrinsic (`julia.simdloop`), which this pass uses to identify loops that were originally marked with `@simd`. Then, this pass looks for chains of floating point operations to mark as fast floating point operations. If the loop was annotated with `ivdep` as well, then the pass marks the loop as having no loop-carried dependencies.
+
+### LowerPTLS
+
+* Filename: `llvm-ptls.cpp`
+* Class Name: `LowerPTLSPass`
+* Opt Name: `module(LowerPTLSPass)`
+
+This pass lowers thread-local Julia intrinsics to assembly instructions. Julia relies on thread-local storage for garbage collection and multithreading task scheduling. When compiling code for system images and package images, this pass replaces calls to intrinsics with loads from global variables that are initialized at load time.
+
+### RemoveAddrspaces
+
+* Filename: `llvm-remove-addrspaces.cpp`
+* Class Name: `RemoveAddrspacesPass`
+* Opt Name: `module(RemoveAddrspaces)`
+
+This pass renames pointers in one address space to another address space. This is used to remove Julia-specific address spaces from LLVM IR.
+
+### RemoveJuliaAddrspaces
+
+* Filename: `llvm-remove-addrspaces.cpp`
+* Class Name: `RemoveJuliaAddrspacesPass`
+* Opt Name: `module(RemoveJuliaAddrspaces)`
+
+This pass removes Julia-specific address spaces from LLVM IR. It is mostly used for displaying LLVM IR in a less cluttered format. Internally, it is implemented off the RemoveAddrspaces pass.
+
+### Multiversioning
+
+* Filename: `llvm-multiversioning.cpp`
+* Class Name: `MultiVersioningPass`
+* Opt Name: `module(JuliaMultiVersioning)`
+
+This pass performs modifications to a module to create functions that are optimized for running on different architectures (see sysimg.md and pkgimg.md for more details). Implementation-wise, it clones functions and applies different target-specific attributes to them to allow the optimizer to use advanced features such as vectorization and instruction scheduling for that platform. It also creates some infrastructure to enable the Julia image loader to select the appropriate version of the function to call based on the architecture the loader is running on. The target-specific attributes are controlled by the `JULIA_CPU_TARGET` environment variable.
```

Comment:
It is controlled by the Module flags `julia.mv.enable` and `julia.mv.specs`, though those values may be influenced by JULIA_CPU_TARGET.